### PR TITLE
refactor(semantic)!: remove name from `reference`

### PR DESCRIPTION
**Pull request by Dunqing**
related to: https://github.com/oxc-project/backlog/issues/32

### CODE REVIEW

**Comment by Dunqing** [comment]
* **#4329** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4329?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by DonIsaac** [comment]
Will this break code on downstream consumers?

**Comment by overlookmotel** [comment]
Nice!

**Comment by Dunqing** [comment]
I checked in `Rolldown`. It doesn't use `reference.name`.

File: crates/oxc_traverse/src/context/scoping.rs [hunk 1]
```diff
@@ -452,16 +450,18 @@ impl TraverseScoping {
...
+        // // Check for unbound references in program with this name
+        // !self.symbols.references.iter().any(|reference| {
+        //     if reference.symbol_id().is_some() {
+        //         // Skip string comparison on bound references, as they'll also be in `symbols.names`
+        //         // which already checked above
+        //         false
+        //     } else {
+        //         reference.name().as_str() == name
+        //     }
+        // })
```
Comment (Dunqing) [inline crates/oxc_traverse/src/context/scoping.rs:464]: Is this for compatibility with Babel's logic? Do we have another way to do it? or just delete it. Seems doesn’t have any tests failed. 

@overlookmotel

Comment (overlookmotel) [reply crates/oxc_traverse/src/context/scoping.rs:464]: I've removed this code in #4356.

**Review by overlookmotel — APPROVED** [review]

**Pull request merged**